File: src/Maximize Number of 1's/Code/solution.java
class Solution {
    public int maxOnes(int arr[], int k) {
        int left = 0;
        int zeroCount = 0;
        int maxLen = 0;

        for (int right = 0; right < arr.length; right++) {
            // Count zeros in the window
            if (arr[right] == 0) {
                zeroCount++;
            }

            // Shrink window if zero count exceeds k
            while (zeroCount > k) {
                if (arr[left] == 0) {
                    zeroCount--;
                }
                left++;
            }

            // Update maximum length
            maxLen = Math.max(maxLen, right - left + 1);
        }

        return maxLen;
    }
}

File: src/Maximize Number of 1's/Code/solution.py
class Solution:
    def maxOnes(self, arr, k):
        left = 0
        zero_count = 0
        max_len = 0

        for right in range(len(arr)):
            # Count zeros
            if arr[right] == 0:
                zero_count += 1

            # Shrink window if zero count exceeds k
            while zero_count > k:
                if arr[left] == 0:
                    zero_count -= 1
                left += 1

            # Update maximum length
            max_len = max(max_len, right - left + 1)

        return max_len
